Lib/zipfile/_path.py

```diff
@@ -281,9 +281,8 @@ def stem(self):
     def filename(self):
         return pathlib.Path(self.root.filename).joinpath(self.at)
 
-    def read_text(self, encoding=None, *args, **kwargs):
-        encoding = io.text_encoding(encoding)
-        with self.open('r', *args, encoding=encoding, **kwargs) as strm:
+    def read_text(self, *args, **kwargs):
+        with self.open('r', *args, **kwargs) as strm:
```

Comment:
If we don't call `io.text_encoding(encoding)` here, `p.read_text()` *is* warned, not caller.

This is current behavior:
```
$ cat zptest.py
import zipfile

zf = zipfile.ZipFile('mysql-master.zip')  # https://github.com/go-sql-driver/mysql/archive/refs/heads/master.zip
p = zipfile.Path(zf, 'mysql-master/README.md')
p.read_text()

$ PYTHONWARNDEFAULTENCODING=1 python3 zptest.py
/Users/inada-n/Downloads/zptest.py:5: EncodingWarning: 'encoding' argument not specified
  p.read_text()
```

Reply:
Ugh.  true.  yet another unfortunate consequence of our limited warning API patterns...

This really should have explicit expected EncodingWarning regression tests for every API that is supposed to raise one. _(annoyingly painful to write)_

I'll need to rework my new test to not use read_text as a proxy for testing behavior in open.  it's late, i'll resume this later.

Reply:
It looks like io.text_encoding takes a stacklevel= arg at least so refactoring this processing out into a common helper method as @jaraco's other comment suggested earlier may be feasible while still getting the desired warning location behavior.

Reply:
EncodingWarning test added.